### PR TITLE
Fix Wrong AppVeyor badge

### PR DESCRIPTION
**Pull request by mattgathu**
this is part of fixing issue #693

- [x] Add an id field with type Option<String> to the Appveyor badge enum variant

- [x]  Add id: alias('badge.attributes.id') to the appveyor badge component

- [x]  Add an image-url attribute, computed from the id to the appveyor badge component that checks to see if id is defined. (I've used `imageUrl` as attribute)

- [x]  Change the appveyor badge template to use the image-url attribute for the img src instead

- [x]  Publish a crate to your local instance that has the URLs dtolnay has listed for serde and manually verify the badge is displayed correctly.

- [x]  Send a pull request to cargo to update the badge documentation to indicate you can specify the appveyor project id if you want to use that instead. (_https://github.com/rust-lang/cargo/pull/4489_)

### CODE REVIEW

**Comment by mattgathu** [comment]
Hi @carols10cents @dtolnay. Kindly review this PR for issue #693

**Comment by carols10cents** [comment]
Hi @mattgathu! Sorry for the delay on reviewing this-- I'm going to be pretty busy today and this weekend prepping for the start of the [impl period](https://internals.rust-lang.org/t/announcing-the-impl-period-sep-18-dec-17/5676), but I will try to get to this next week :) Thank you for your PR!!

File: app/components/badge-appveyor.js [hunk 1]
```diff
@@ -6,8 +6,22 @@ export default Component.extend({
     tagName: 'span',
     classNames: ['badge'],
 
+    id: alias('badge.attributes.id'),
     repository: alias('badge.attributes.repository'),
 
+    imageUrl: computed('badge.attributes.id', function() {
+        let id = this.get('badge.attributes.id');
+        let branch = this.get('branch');
+        if (id === undefined || id === null) {
```
Comment (carols10cents) [inline app/components/badge-appveyor.js]: I think this condition is backwards? This says if id IS undefined or id IS null, then try to use id? This should be `(id !== undefined && id !== null)`, yeah? I tried making a badge using only `repository` and `id` and the `id` isn't appearing in the image url :(

**Comment by jtgeibel** [comment]
@mattgathu I think the operator also needs to be switched from `||` to `&&`.

**Comment by mattgathu** [comment]
@carols10cents  @jtgeibel thank you for the bug catch! I've fixed it.

**Comment by carols10cents** [comment]
Thank you!!! Works great now!!! 

bors: r+

**Pull request merged**